tests/arch/arm/arm_thread_swap/src/arm_thread_arch.c

```diff
@@ -37,7 +37,7 @@
 #define FPSCR_MASK		(0xffffffffU)
 #endif
 
-extern void z_move_thread_to_end_of_prio_q(struct k_thread *thread);
+extern void z_yield_testing_only(struct k_thread *curr);
```

Comment:
`extern` isn't necessary here but not blocking 

```suggestion
void z_yield_testing_only(struct k_thread *curr);
```